badgecheck/verifier.py

```diff
@@ -51,10 +55,24 @@ def verify(badge_input):
         task_meta = active_tasks[0]
         task_func = tasks.task_named(task_meta['name'])
 
-        if task_meta['id'] == last_task_id:
+        if task_meta['task_id'] == last_task_id:
             break
 
-        last_task_id = task_meta['id']
+        last_task_id = task_meta['task_id']
         call_task(task_func, task_meta, store)
 
-    return store.get_state()
+    state = store.get_state()
+    failed_tasks = filter_failed_tasks(state)
+    ret = {
+        'messages': [],
+        'graph': state['graph'],
+        'input': state['input']
+    }
+    for task in failed_tasks:
+        ret['messages'].append(format_message(task))
+
+    ret['errorCount'] = len([m for m in ret['messages'] if m['messageLevel'] == MESSAGE_LEVEL_ERROR])
```

Comment:
Minor: You could use a counter to avoid walking through tasks/messages multiple times?

```
from collections import Counter
...
count = Counter()
for task in failed_tasks:
    message = format_message(task)
    ret['messages'].append(message)
    count[message['messageLevel']] += 1

ret['errorCount'] = count[MESSAGE_LEVEL_ERROR]
ret['warningCount'] = count[MESSAGE_LEVEL_WARNING]
ret['valid'] = not bool(ret['errorCount'])
```